src/test/java/org/folio/auth/authtokenmodule/AuthTokenTest.java

```diff
@@ -1023,4 +1043,161 @@ private String getMagicPermission(String endpoint) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    var rt = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    ts.saveToken(rt)
+      .compose(x -> ts.checkTokenNotRevoked(rt))
+      .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testStoreRefreshTokenNotFound(TestContext context) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    // A RefreshToken which doesn't exist is treated as revoked.
+    var unsavedToken = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    ts.checkTokenNotRevoked(unsavedToken).onComplete(context.asyncAssertFailure());
+  }
+
+  @Test
+  public void testStoreSaveApiToken(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken).compose(x ->  ts.checkTokenNotRevoked(apiToken))
+      .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testStoreApiTokenNotFound(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    // A ApiToken which doesn't exist in storage is treated as revoked.
+    var unsavedToken = new ApiToken(tenant);
+    ts.checkTokenNotRevoked(unsavedToken).onComplete(context.asyncAssertFailure());
+  }
+
+  @Test
+  public void testApiTokenRevoked(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken)
+      .compose(x -> ts.checkTokenNotRevoked(apiToken))
+      .compose(x -> ts.revokeToken(apiToken))
+      .onComplete(context.asyncAssertSuccess())
+      .compose(x -> ts.checkTokenNotRevoked(apiToken))
+      .onComplete(context.asyncAssertFailure());
+  }
+
+  @Test
+  public void testStoreRefreshTokenSingleUse(TestContext context) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    // Create and save some tokens.
+    var rt1 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var rt2 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var rt3 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var s1 = ts.saveToken(rt1);
+    var s2 = ts.saveToken(rt2);
+    var s3 = ts.saveToken(rt3);
+
+    Async async = context.async();
+    CompositeFuture.all(s1, s2, s3)
+      .compose(a -> ts.checkTokenNotRevoked(rt2))
+      .onComplete(context.asyncAssertSuccess()) // First check should succeed.
+      .compose(a -> ts.checkTokenNotRevoked(rt2))
+      .onComplete(context.asyncAssertFailure(b ->  { // Second should fail.
+        // At this point all tokens for the user should be revoked so any subsequent
+        // checks should fail.
+        ts.checkTokenNotRevoked(rt1).onComplete(context.asyncAssertFailure(c -> {
+          ts.checkTokenNotRevoked(rt3).onComplete(context.asyncAssertFailure(d -> {
+            async.complete();
+          }));
```

Comment:
```
          context.asyncAssertFailure(d -> {
            async.complete();
          })
```
can always be replaced by
```
          context.asyncAssertFailure()
```

Reply:
I'm finding that without `async.complete()` these test methods don't complete and time out.

Reply:
Noticing that it is only when I do `CompositeFuture.all()` that I need `async.complete()`.